apps/src/templates/utils/expandableImages.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {ImagePreview} from '../instructions/AniGifPreview';
+
+/**
+ * A basic helper function to enable the "expandable images" we use in our
+ *     curriculum content.
+ *
+ * @param {Element} node - The DOM node containing the images. Supports nodes
+ *     with multiple contained images.
+ * @param {function} showImageDialog - The method to call when the user clicks
+ *     on the image, which will show the modal dialog containing the expanded
+ *     image.
+ *
+ * @see @cdo/apps/src/templates/plugins/expandableImages.js
+ * @see @cdo/apps/src/redux/instructionsDialog.js
+ */
+export function renderExpandableImages(node, showImageDialog) {
+  const expandableImages = node.querySelectorAll('.expandable-image');
+  for (let i = 0; i < expandableImages.length; i++) {
+    const expandableImg = expandableImages[i];
+
+    /*
+     * TODO: When this method is used in the context of an existing React
+     * component, it raises the following warning:
+     *
+     * > Replacing React-rendered children with a new root component. If you
+     * > intended to update the children of this node, you should instead have
+     * > the existing children update their state and render the new components
+     * > instead of calling ReactDOM.render.
+     *
+     * We should probably rebuild this in such a way as to not violate React's
+     * expectations like this.
```

Comment:
I think I am missing something here. the way this is implemented in this PR, do we always hit this warning? If not, what are the criteria for calling this function without triggering this warning?

Reply:
Yes, we always hit this warning. Note that this PR is not implementing this functionality, it's just reorganizing it; we _currently_ always hit this warning when rendering expandable images in the context of React. (For historical context, expandable images were originally implemented outside the context of React).

In the interest of aligning to the "do no harm" approach to tech debt we've settled on, I decided not to fix the implementation as part of this work but instead to add this comment. Although, we should probably also add a work item somewhere to try to get that work on an actual schedule somewhere. Any suggestions for where/when?

Reply:
Thanks Elijah, all makes sense. This could go into the Q2 Tech Debt sprint in Jira.